### ISSUE TITLE
Enhancing Hyperparameter Study Parallelization with Batch Update to Expected Improvement Function

### ISSUE THREAD

**Issue by adendek** [optuna/optuna#5610]
### Motivation

In exploring the Tree-structured Parzen Estimator (TPE) implementation for insights on improving study parallelization, I've identified a potential enhancement beneficial for my project, which utilizes this tool for optimizing gradient boosting tree (GBT) hyperparameters. Despite our dataset being relatively small (on the order of tens of MB), we train a significant number of models (on the order of hundreds) for each we would like to preform hyperpareter optimization. Leveraging a serverless-like platform like [Azure Container Apps](https://azure.microsoft.com/en-us/products/container-apps), we have the capability to concurrently train up to 1,000 trials. However, it appears there's no existing method to run multiple trials in parallel while still benefiting from TPE or Bayesian optimization techniques.

### Description

I propose a strategy to facilitate study parallelization as follows:

* Initial Parallel Trials: Execute the initial `n_trials` concurrently, leveraging their random and independent nature.
* Batch Sampling Modification: Adapt the sampling method to select the `top_k` values (where k equals `batch_size`) instead of a single maximum [value](https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/sampler.py#L537), enhancing the efficiency of trial selection: 
```python
# current approach
best_idx = np.argmax(acquisition_func_vals)
# proposed approach 
best_idxs = top_k(acquisition_func_vals, batch_size)
```

The top_k implementation can reference [this Stack Overflow answer](https://stackoverflow.com/a/23734295/1877600)..
* Parallel Experiment Execution: Conduct `batch_size` experiments in parallel, optimizing resource utilization.
* Batch Updates to Sampling Function: Implement batch updates, allowing for more efficient sampling and selection processes.
* Execution Backend Specification (Optional): Provide users with the option to specify the execution backend, enhancing flexibility.

### Additional context (optional)

I am prepared to spearhead the implementation of this proposal. However, before commencing, I seek to confirm whether this concept is in alignment with Optuna's development roadmap and if it would be a welcome contribution.
